### PR TITLE
667 - include endorsed block in endorsing rights

### LINKED ISSUE #667: Include Endorsed Block in Endorsing Rights
Add the `level` to `endorsing_rights` to be able to show the "Endorsed Block" without querying the `operations` endpoint with `kind: endorsement`.

`null` for future rights.

### PR DESCRIPTION
**Pull request by piotrkosecki**
resolves #667 

## required DB changes
In table `endorsing_rights` new column needs to be added:
```
    endorsed_block integer
```

### CODE REVIEW

**Comment by vishakh** [comment]
Can you describe the nature of the change and the algorithm you used here?

**Comment by piotrkosecki** [comment]
@vishakh Data required to update `endorsing_rights` table with the `endorsed_block` is in the operations - `block.operationGroups.content`(simplified a bit because operationGroups and content are lists). I look for operation type `Endorsment` and if delegate in operation matches the one in the endorsing rights I take the `level` of this operation and update `endorsing_rights` with it.

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 1]
```diff
@@ -390,7 +390,24 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
...
+                      case (block, _) =>
+                        fetchRights.blockHash.contains(block.data.hash)
+                    }.flatMap {
+                      case (block, _) =>
+                        block.operationGroups.flatMap {
+                          _.contents.collect {
+                            case e: Endorsement if e.metadata.delegate.value == rights.delegate => e
+                          }.map(_.level)
+                        }.headOption
+                    }
```
[the remark this answers is not in the document]
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: ok, extracted

File: src/main/scala/tech/cryptonomic/conseil/Lorre.scala [hunk 2]
```diff
@@ -390,7 +390,24 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
...
+                    rights.copy(endorsedBlock = endorsedBlock)
+                  }
+          }
+          (db.run(TezosDb.upsertBakingRights(br)), db.run(TezosDb.upsertEndorsingRights(updatedEndorsingRights)))
             .mapN((_, _) => ())
```
[the remark this answers is not in the document]
Comment (piotrkosecki) [reply src/main/scala/tech/cryptonomic/conseil/Lorre.scala]: sure, done

**Pull request merged**